integrations/jenkins/client.py

```diff
@@ -0,0 +1,103 @@
+from typing import Any, AsyncGenerator, Optional
+from urllib.parse import urlparse
+import httpx
+
+from loguru import logger
+
+from port_ocean.context.event import event
+from port_ocean.utils import http_async_client
+
+
+class JenkinsClient:
+    def __init__(
+        self, jenkins_base_url: str, jenkins_user: str, jenkins_token: str
+    ) -> None:
+        self.jenkins_base_url = jenkins_base_url
+        self.client = http_async_client
+        self.client.auth = httpx.BasicAuth(jenkins_user, jenkins_token)
+
+    async def get_all_jobs_and_builds(
```

Comment:
why do u have 2 functions with the same purpose?

Reply:
I can refactor and simplify them.